src/core/nodeOps.ts

```diff
@@ -184,6 +184,14 @@ export const nodeOps: RendererOptions<TresObject, TresObject> = {
       let finalKey = kebabToCamel(key)
       let target = root?.[finalKey]
 
+      if (key === 'args' && !deepArrayEqual(_prevValue, nextValue)) {
+        const prevNode = node as TresObject3D
+
+        if (node.type && nextValue.length > 0) {
+          root = Object.assign(prevNode, new catalogue.value[node.type](...nextValue))
+        }
+      }
+
```

Comment:
I think there should be premature return, or just add a condition on L220

Reply:
Done